### PR TITLE
hubble/metrics: Add workload-name and app options to sourceContext and destinationContext

### PR DESCRIPTION
**Pull request by chancez**
```release-note
hubble/metrics: Add workload-name and app options to sourceContext and destinationContext
```

### CODE REVIEW

**Comment by chancez** [comment]
PR is currently based on https://github.com/cilium/cilium/pull/21181

**Review by kaworu — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Comment by kaworu** [comment]
[**ci-datapath** failed](https://github.com/cilium/cilium/actions/runs/3176599187/jobs/5176097750) with:

```
Unable to find image 'registry:2' locally
docker: Error response from daemon: Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
See 'docker run --help'.
Error: Process completed with exit code 125.
```

Should be fixed by https://github.com/cilium/cilium/pull/21525 which has been merged, @chancez could you rebase this PR against `master` please?

File: pkg/hubble/metrics/api/context.go [hunk 1]
```diff
@@ -40,14 +40,18 @@ const (
 	// purpose. It uses "reserved:kube-apiserver" label if it's present in the identity label list.
 	// Otherwise, it uses the first label in the identity label list with "reserved:" prefix.
 	ContextReservedIdentity
+	// ContextWorkloadName uses the pod's workload name for identification.
+	ContextWorkloadName
+	// ContextWorkloadName uses the pod's app label for identification.
```
Comment (aditighag) [inline pkg/hubble/metrics/api/context.go]:
```suggestion
	// ContextApp uses the pod's app label for identification.
```

**Review by aditighag — APPROVED** [review]
Minor comment to fix the typo.

File: pkg/hubble/metrics/api/context.go (hunk 1 above)
Comment (chancez) [reply pkg/hubble/metrics/api/context.go]: Good catch. Fixed.

**Comment by chancez** [comment]
/test-1.16-4.9

**Pull request merged**